client/llb/exec.go

```diff
@@ -624,6 +640,12 @@ func AddUlimit(name UlimitName, soft int64, hard int64) RunOption {
 	})
 }
 
+func AddCDIDevice(name string) RunOption {
+	return runOptionFunc(func(ei *ExecInfo) {
+		ei.State = ei.State.AddCDIDevice(name)
```

Comment:
This needs to take at least the required bool. The string value should be `kind | kind=device | classifier`. I'm not sure if it makes sense to break these apart in the LLB level or keep as a single string variable.

Reply:
> The string value should be `kind | kind=device | classifier`. I'm not sure if it makes sense to break these apart in the LLB level or keep as a single string variable.

I think better to keep name as a string as we are already validating it when generating the spec: https://github.com/moby/buildkit/pull/4056/files#diff-478a239cda41c7aff441eef90d0eae4fc531f3fcc30db204e9dfbe64b9d863aaR182-R184